api/envoy/extensions/filters/http/compressor/v3/compressor.proto

```diff
@@ -123,4 +123,7 @@ message Compressor {
   //    instead of
   //    ``<stat_prefix>.compressor.<compressor_library.name>.<compressor_library_stat_prefix>.*``.
   ResponseDirectionConfig response_direction_config = 8;
+
+  // If true, chooses this compressor first to do compression when the q-values are same.
+  bool choose_first = 9;
```

Comment:
Not expert, but few questions here:

* What happened if multiple compressor filter has choose_first as true?
* Is there a case people want to specify different `choose_first` for request and response direction?
* How about adding weight for each compressor filter instead of just first choose?

Reply:
1. will choose the last compressor which enables `choose_first` if multiple compressor filter has choose_first as true.
2. `choose_first` affects the decision of `Accept-Encoding` that means only response direction needs this.

Reply:
3.I think it is not common case because users usually prefer to use better unique compressor instead of "one gzip one brotli".

Reply:
Thanks for the reply

> 1. will choose the last compressor which enables `choose_first` if multiple compressor filter has choose_first as true.
Could you describe this behavior in the API comment?

> 2. `choose_first` affects the decision of `Accept-Encoding` that means only response direction needs this.
Got it, thanks, sorry for the silly question.
> 3.I think it is not common case because users usually prefer to use better unique compressor instead of "one gzip one brotli".
Yea, agree that weight doesn't make sense here.

Reply:
> Thanks for the reply
> 
> > 1. will choose the last compressor which enables `choose_first` if multiple compressor filter has choose_first as true.
> 
> Could you describe this behavior in the API comment?
> 

OK, thanks!